File: ecs.ts
import { ComponentT, SchemaT } from './bimrepo';

export class ECSComponent {

    private simplifiedName: string;

    constructor(name: string)
    {
        this.simplifiedName = name;
    }

    getSimplifiedName()
    {
        return this.simplifiedName;
    }

    exportToDataArray() : ComponentT
    {
        return new ComponentT();
    }
    
    exportDefinitionToArray(): SchemaT {
        return new SchemaT();
    }
}

File: ecs.cs

using bimrepo;

public class ECSComponent {

    private string simplifiedName;

    public ECSComponent(string name)
    {
        this.simplifiedName = name;
    }

    public string getSimplifiedName()
    {
        return this.simplifiedName;
    }

    public virtual ComponentT exportToDataArray()
    {
        return new ComponentT();
    }

    public virtual SchemaT exportDefinitionToArray() {
        return new SchemaT();
    }
}

